libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayProgressObserver.kt

```diff
@@ -11,21 +11,22 @@ import com.mapbox.navigation.core.trip.session.RouteProgressObserver
  * This class will feed locations to your [MapboxReplayer] and simulate
  * your active route for you.
  */
-class ReplayProgressObserver(
+class ReplayProgressObserver @JvmOverloads constructor(
     /**
      * As navigation receives [RouteProgress], this will push events to your
      * replay history player to be played.
      */
-    private val mapboxReplayer: MapboxReplayer
+    private val mapboxReplayer: MapboxReplayer,
+    private val replayRouteMapper: ReplayRouteMapper = ReplayRouteMapper()
 ) : RouteProgressObserver {
 
-    private val replayRouteMapper = ReplayRouteMapper()
     private var currentRouteLeg: RouteLeg? = null
 
     /**
      * @param options allow you to control the driver and car behavior.
      * @return [ReplayProgressObserver]
      */
+    @Deprecated("Provide a mapper through construct and change its options when needed.")
```

Comment:
This is unrelated to a leak right?

Reply:
what's the reason for deprecating here? this seems to only remove the ability to change the replay speed, but the pr makes it appear as if it has something to do with memory leaks 🤔